lib/psci/psci_off.c

```diff
@@ -119,10 +119,9 @@ int psci_do_cpu_off(unsigned int end_pwrlvl)
 #endif
 
 	/*
-	 * Arch. management. Perform the necessary steps to flush all
-	 * cpu caches.
+	 * Arch. management. Initiate power down sequence.
 	 */
-	psci_do_pwrdown_cache_maintenance(psci_find_max_off_lvl(&state_info));
```

Comment:
@nmenon, This is a problem that needs to be solved between the CPU- Operations framework and the platform. If you take a look at the CPU- operations that are currently implemented, all of the CPUs only implement till Level 2 cache and hence they flush only till Level 2 in their cluster power down sequence. So , in you case, if the Level 3 cache is part of the CPU Architecture, then you would do the cache flush for the same in the corresponding CPU-operations for that CPU.

Otherwise, if the additional level of Cache is part of the platform (eg: system caches in interconnect), then you would do the L3 cache flush in the platform hooks for the particular platform.

Hope that helps.